pkg/volume/gce_pd/gce_util.go

```diff
@@ -122,6 +122,167 @@ func (util *GCEDiskUtil) DetachDisk(pd *gcePersistentDisk) error {
 	if err := gce.(*gce_cloud.GCECloud).DetachDisk(pd.pdName); err != nil {
 		return err
 	}
+
+	// Verify disk detached, retry if needed.
+	go verifyDiskDetachedRetryIfNeeded(pd, gce)
+	return nil
+}
+
+// Verifys the disk device to be created has been succesffully attached, and retries if it fails.
+func verifyDiskAttachedRetryIfNeeded(pd *gcePersistentDisk, sdBeforeSet util.StringSet, gce cloudprovider.Interface) (string, error) {
+	devicePaths := getDiskByIdPaths(pd)
+	for numRetries := 0; numRetries < 10; numRetries++ {
+		for numChecks := 0; numChecks < 10; numChecks++ {
+			if err := udevadmChangeToNewDrives(sdBeforeSet); err != nil {
+				// udevadm errors should not block disk attachment, log and continue
+				glog.Errorf("%v", err)
+			}
+
+			for _, path := range devicePaths {
+				if pathExists, err := pathExists(path); err != nil {
+					return "", err
+				} else if pathExists {
+					// A device path has succesfully been created for the PD
+					glog.V(5).Infof("Succesfully attached GCE PD %q.", pd.pdName)
+					return path, nil
```

Comment:
The logic here is when a disk is attached there are multiple symlinks that are created--all of them link back to the same `/dev/sd*` drive. We watch all the possible symlinks and declare success when any one of them show up. This is not a problem, because we just need one handle to the PD and whichever symlink shows up first is sufficient for the PD to be considered attached and it can be used as the as the global mount point.

More robust to watch all of them, and grab on to the first one that shows up. And then as long as one shows up we know that the device is successfully attached and can use that as the global mount point.